client/app/components/OnboardingGuide/OnboardingGuide.js

```diff
@@ -9,87 +9,58 @@ import GuideFilterPage from './GuideFilterPage';
 // Select child component (page/view) to be rendered
-// Returns object (including child component) based on props.data & nextStep
-const selectChild = function selectChild(data, nextStep) {
-  const { path, onboarding_data } = data;
-  const pageData = (path.length > 0) ?
-    _.find(onboarding_data, (pd) => pd.sub_path === path.substring(1)) :
-    {};
-
-  switch (path) {
-    case '/slogan_and_description':
+// Returns object (including child component) based on props.data
+const selectChild = function selectChild(data) {
+  const { page, onboarding_data } = data;
+  const pageData = _.find(onboarding_data, (pd) => pd.step === page) || {};
+
+  switch (page) {
+    case 'slogan_and_description':
       return { Page: GuideSloganAndDescriptionPage, pageData };
-    case '/cover_photo':
+    case 'cover_photo':
       return { Page: GuideCoverPhotoPage, pageData };
-    case '/filter':
+    case 'filter':
       return { Page: GuideFilterPage, pageData };
-    case '/paypal':
+    case 'paypal':
       return { Page: GuidePaypalPage, pageData };
-    case '/listing':
+    case 'listing':
       return { Page: GuideListingPage, pageData };
-    case '/invitation':
+    case 'invitation':
       return { Page: GuideInvitationPage, pageData };
     default:
-      return { Page: GuideStatusPage, onboarding_data, nextStep };
+      return { Page: GuideStatusPage, onboarding_data };
   }
 };
 
-// Get link and title of next recommended onboarding step
-const nextStep = function nextStep(data) {
-  const nextStepData = data.find((step) => !step.complete);
-
-  const titles = {
-    slogan_and_description: 'web.admin.onboarding.guide.next_step.slogan_and_description',
-    cover_photo: 'web.admin.onboarding.guide.next_step.cover_photo',
-    filter: 'web.admin.onboarding.guide.next_step.filter',
-    paypal: 'web.admin.onboarding.guide.next_step.paypal',
-    listing: 'web.admin.onboarding.guide.next_step.listing',
-    invitation: 'web.admin.onboarding.guide.next_step.invitation',
-  };
-
-  if (nextStepData) {
-    return {
-      title: t(titles[nextStepData.step]),
-      link: nextStepData.sub_path,
-    };
-  } else {
-    return null;
-  }
-};
+function setPushState(state, title, path) {
 
-// getPaths: initial path containing given pathFragment & relative (deeper) path
-const getPaths = function getPaths(props, pathFragment) {
-  const pathParts = props.data.original_path.split(pathFragment);
-  const initialPath = pathParts[0] + pathFragment;
-  return { initialPath, componentSubPath: pathParts[1] };
-};
+  // React has an internal variable 'canUseDOM', which we emulate here.
+  const canUseDOM = !!(typeof window !== 'undefined' &&
+                        window.document &&
+                        window.document.createElement);
+  const canUsePushState = !!(typeof history !== 'undefined' &&
+                              history.pushState);
+
+  if (canUseDOM && canUsePushState) {
+    window.history.pushState(state, title, path);
+  }
+}
```

Comment:
Definitely. I kind of started doing something like that but then reverted, because I got a bit stuck + got the feeling that this PR is large enough already.